labs/lab2/cord.cpp

```diff
@@ -49,19 +49,48 @@ cord_t cord_join(cord_t R, cord_t S) {
 
 string cord_tostring(cord_t R) {
     // TODO: Your implementation here
-    return "";
+    if (!R->len)
+        return string();
+    if (R->data.length())
```

Comment:
Yes you need to distinguish leaf node and non-leaf node. But according to the README, 
> The data field of a non-leaf is unspecified.

So you need to follow the definition and check whether a valid cord has `nullptr` left and right fields.
```suggestion
    if (R->left == nullptr && R->right == nullptr)
```